src/dialog-controller.ts

```diff
@@ -102,7 +162,9 @@ export class DialogController {
 
     const dialogResult: DialogCloseResult = { wasCancelled: !ok, output };
 
-    return this.closePromise = invokeLifecycle(this.controller.viewModel || {}, 'canDeactivate', dialogResult)
+    // tslint:disable-next-line:max-line-length
+    // TODO: add canDeactivate API
```

Comment:
Still in process?

Reply:
Leftover, I decided not to do it in this PR.